kview/src/main/java/dev/ikm/komet/kview/mvvm/viewmodel/PatternViewModel.java

```diff
@@ -307,43 +311,47 @@ public boolean createPattern() {
+                        .purpose(((EntityFacade)getPropertyValue(PURPOSE_ENTITY)).toProxy());
+            patternAssembler.attach((FullyQualifiedName fqn) -> fqn
+                                    .language(((EntityFacade)getPropertyValue(FQN_LANGUAGE)).toProxy())
+                                    .text(getPropertyValue(FQN_DESCRIPTION_NAME_TEXT))
+                                    .caseSignificance(((EntityFacade)getPropertyValue(FQN_CASE_SIGNIFICANCE)).toProxy()));
             // add the field definitions
             for (int i = 0; i< fieldsProperty.size(); i++) {
                 PatternField patternField = fieldsProperty.get(i);
                 patternAssembler.fieldDefinition(patternField.meaning().toProxy(), patternField.purpose().toProxy(),
                         patternField.dataType().toProxy(), i);
             }
-            patternAssembler.purpose(((EntityFacade)getPropertyValue(PURPOSE_ENTITY)).toProxy())
-                            .attach((FullyQualifiedName fqn) -> fqn
-                                    .language(((EntityFacade)getPropertyValue(FQN_LANGUAGE)).toProxy())
-                                    .text(getPropertyValue(FQN_DESCRIPTION_NAME_TEXT))
-                                    .caseSignificance(((EntityFacade)getPropertyValue(FQN_CASE_SIGNIFICANCE)).toProxy()));
         });
 
         // add the other name description semantics if they exist
         ObservableList<DescrName> otherNamesProperty = getObservableList(OTHER_NAMES);
-        if (!otherNamesProperty.isEmpty()) {
-            otherNamesProperty.forEach(otherName ->
-                    session.compose(new Synonym()
-                                    .language(otherName.getLanguage().toProxy())
-                                    .text(otherName.getNameText())
-                                    .caseSignificance(otherName.getCaseSignificance().toProxy()), pattern)
-                            .attach(new USDialect()
-                                    .acceptability(ACCEPTABLE))
-            );
-        }
+        boolean isEdit = getPropertyValue(MODE).equals("EDIT");
+
+        otherNamesProperty.forEach(otherName -> {
+            Synonym synonym = new Synonym()
+                    .language(otherName.getLanguage().toProxy())
+                    .text(otherName.getNameText())
+                    .caseSignificance(otherName.getCaseSignificance().toProxy());
+            if (isEdit) {
+                HashMap<DescrName, SemanticEntityVersion> regularNamesMap = getPropertyValue(REGULAR_NAMES);
+                if(regularNamesMap != null && regularNamesMap.get(otherName) !=null) {
+                    SemanticEntityVersion semanticEntityVersion = regularNamesMap.get(otherName);
+                    SemanticEntity<SemanticEntityVersion> semanticEntity = semanticEntityVersion.chronology();
+                    synonym.semantic(semanticEntity.toProxy());
+                }
+            }
```

Comment:
@swaroopsalvi Can we have @dholubek weigh in on this? Things look good but I feel there's some redundancy in `OTHER_NAMES` and `REGULAR_NAMES` w.r.t the array list and map respectively.
(purely optional) - In addition, we could rename the `REGULAR_NAMES` to `OTHER_NAME_SEMANTIC_VERSION_MAP`.

Reply:
@carldea , Doug and I spoke briefly and explained him why  I implemented the way I have currently. This can be improvised for sure but then it will be better to have it as part of the bigger refactor effort since DescrName class is used at multiple places in Pattern editing.

Reply:
@carldea I talked with @swaroopsalvi and this was to avoid refactoring and breaking create pattern.  I created a story to refactor and combine the collections and use the hashmap one going forward.  
https://ikmdev.atlassian.net/browse/IIA-1166